### PR TITLE
CAPI+CAPA provider upgrade docs

### LINKED ISSUE #4268: Provider version upgrades reset feature gates
/kind bug

**What steps did you take and what happened:**

Whilst upgrading from `v2.0.2` to `v2.1.1`, various feature gates that were set for things like MachinePools and role additions were reset to `false`.

**What did you expect to happen:**

On first thought - I didn't expect the feature gates to be reset on a version upgrade. On second thought, however - it made sense. Since the upgrade path removes older version `Deployments` and creates new ones, there's no way for it to know which feature gates should be carried forward to the new version. This can be done by setting feature gate environment variables, but I suppose it wasn't entirely obvious this was needed. 

**Anything else you would like to add:**

To that end - raising a PR that adds a doc to the CAPA book, listing prerequisites and steps for carrying out provider upgrades.

**Environment:**

- Cluster-api-provider-aws version: `v2.0.2 -> v2.1.1`
- Kubernetes version: (use `kubectl version`): `Client Version: version.Info{Major:"1", Minor:"26", GitVersion:"v1.26.1", GitCommit:"8f94681cd294aa8cfd3407b8191f6c70214973a4", GitTreeState:"clean", BuildDate:"2023-01-18T15:58:16Z", GoVersion:"go1.19.5", Compiler:"gc", Platform:"darwin/amd64"}`
- OS (e.g. from `/etc/os-release`): macOS 12.5

### PR DESCRIPTION
**Pull request by rudimk**
**What type of PR is this?**

/kind documentation
**What this PR does / why we need it**:

At the moment, we don't have a doc in the CAPA book that provides an overview of prerequisites needed for upgrading CAPA versions, as well as the actual upgrade process itself. Note - this PR supersedes #4269 - that PR's branch on my end got fouled up. cc @Ankitasw @Skarlso 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #4268 

**Special notes for your reviewer**:

**Checklist**:

- [X] squashed commits
- [X] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

**Release note**:

```release-note
Add documentation to the Cluster API AWS book on running provider version upgrades.
```

### CODE REVIEW

File: docs/book/src/topics/provider-upgrades.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Upgrading the Cluster API AWS Provider
+
+The `clusterctl` tool ships with an upgrade path designed to check for available updates and apply a contract to upgrade a CAPA management cluster to the latest version. The upgrade path will download manifests for the latest versions of Cluster API and the AWS provider, apply those to your management cluster and scale down the `Deployments` running older versions.
+
+During the upgrade, clusters provisioned by CAPA will continue to function normally. Still, the ability to use the management cluster to provision new clusters/manage or reconcile existing clusters will be restricted. For production workloads, it's best to upgrade your CAPA management cluster when you know you won't need to create or modify existing clusters.
```
Comment (AverageMarcus) [inline docs/book/src/topics/provider-upgrades.md:5]: Slight change in wording as sounded strange to me. Very nitpick-y.
```suggestion
During the upgrade, clusters provisioned by CAPA will continue to function normally. Still, the ability to use the management cluster to provision new clusters or manage / reconcile existing clusters will be restricted. For production workloads, it's best to upgrade your CAPA management cluster when you know you won't need to create or modify existing clusters.
```

**Review by AverageMarcus — COMMENTED** [review]
/lgtm

Just small nitpick on some wording.

File: docs/book/src/topics/provider-upgrades.md (hunk 1 above)
Comment (rudimk) [reply docs/book/src/topics/provider-upgrades.md:5]: Thannks for flagging this - pushed in a squashed commit with the tweak.

**Review by AverageMarcus — COMMENTED** [review]
/lgtm

**Comment by Ankitasw** [comment]
@rudimk could you please squash your commits, as bot says this PR contains merge commits, which should not happen.

**Comment by rudimk** [comment]
@Ankitasw thanks for the reminder! apologies - i've been away for a few weeks and completely missed this. as (bad) luck would have it, i might have to redo this PR anyway since i've managed to foul up my local clone. luckily, found the git process guide we're supposed to use so i'll probably close this PR in a day or two and push a proper one up(again!).

**Comment by richardcase** [comment]
Not for this PR but just testing the job:

/test pull-cluster-api-provider-aws-e2e-eks

**Comment by richardcase** [comment]
@rudimk - would you be able to rebase/squash? I thought I could get around it, but sadly not. Please reach out if you need a hand.

**Comment by rudimk** [comment]
Apologies for leaving this hanging. I've tried rebasing and squashing but I end up fouling my fork/branch every time. Turns out all I needed to do was read the official Git workflow docs here: 
https://github.com/kubernetes/community/blob/master/contributors/guide/github-workflow.md#4-keep-your-branch-in-sync. I'll take care of this over the next 1-2 days for sure, have it ready to be reviewed by Friday.

**Pull request closed without merging**